### PR TITLE
Add documentation covering usage of the test-proxy config feature

### PR DESCRIPTION
**Pull request by scbedd**
@samvaity for FYI, totally missed doing so with the original PR.

### CODE REVIEW

**Review by samvaity — APPROVED** [review]

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 1]
```diff
@@ -149,32 +149,70 @@ For safety, the "official target" version that the azure-sdk team uses is presen
...
-  restore    Restore the assets, referenced by assets.json, from git.
+/>Azure.Sdk.Tools.TestProxy --help
+Description:
+  This tool is used by the Azure SDK team in two primary ways:
+
+    - Run as a http record/playback server. ("start" / default verb)
+    - Invoke a CLI Tool to interact with recordings in an external store. ("push", "restore", "reset", "config")
+
+Usage:
+  Azure.Sdk.Tools.TestProxy [command] [options]
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:174]: Should this still be `test-proxy` assuming most are encouraged to use the standalone executable?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 2]
```diff
@@ -149,32 +149,70 @@ For safety, the "official target" version that the azure-sdk team uses is presen
...
+
+Options:
+  -l, --storage-location <storage-location>  The path to the target local git repo. If not provided as an argument, Environment
+                                             variable TEST_PROXY_FOLDER will be consumed. Lacking both, the current working
+                                             directory will be utilized. []
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:179]: Trailing `[]`?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 3]
```diff
@@ -149,32 +149,70 @@ For safety, the "official target" version that the azure-sdk team uses is presen
...
+  -p, --storage-plugin <storage-plugin>      The plugin for the selected storage, default is Git storage is GitStore. (Currently
+                                             the only option) [default: GitStore]
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:181]: Is this line break going to show up in the output?

File: tools/test-proxy/documentation/asset-sync/README.md [hunk 4]
```diff
@@ -103,6 +105,30 @@ After assets have been restored and then modified (re-recorded etc.) a push will
...
+
+- `locate`
+- `show`
+
+`locate`: Dumps which folder under the `.assets` folder contains your recordings. [See `config` usage in layout section below.](#layout-within-a-language-repo)
+`show`: Dumps the contents of the targeted assets.json.
+
+```bash
+# from C:/repo/azure-sdk-for-python, the root of the python language repository
+test-proxy config locate --assets-json-file sdk/keyvault/azure-keyvault-keys/assets.json
```
Comment (benbp) [inline tools/test-proxy/documentation/asset-sync/README.md:124]: Remind me, does this work without `--assets-json-file` if you're in a directory with an `assets.json` file?

**Review by benbp — APPROVED** [review]
Some minor stuff, but LGTM overall

File: tools/test-proxy/documentation/asset-sync/README.md (hunk 4 above)
Comment (scbedd) [reply tools/test-proxy/documentation/asset-sync/README.md:124]: No currently.

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:174]: It's not here?

Comment (benbp) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:174]: Yeah but if you have this installed standalone or via dotnet tool won't `test-proxy --help` print out `Azure.Sdk.Tools.TestProxy` as the example command with this change?

**Pull request merged**